crates/red_knot/src/symbols.rs

```diff
@@ -523,11 +540,72 @@ where
     }
 }
 
+pub(crate) struct ReachableDefinitionsIterator<'a> {
+    table: &'a SymbolTable,
+    flow_node_id: FlowNodeId,
+    symbol_id: SymbolId,
+}
+
+impl<'a> Iterator for ReachableDefinitionsIterator<'a> {
+    type Item = Definition;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            match &self.table.flow_graph.flow_nodes_by_id[self.flow_node_id] {
+                FlowNode::Start => return None,
+                FlowNode::Definition(def_node) => {
+                    self.flow_node_id = def_node.predecessor;
+                    if def_node.symbol_id == self.symbol_id {
+                        return Some(def_node.definition.clone());
```

Comment:
We should look into making `Definition` `Copy` if that's something we return frequently. I add it to our monthly plan.